File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
      vector<int> ans;
    int kthSmallest(TreeNode* root, int k) {
        traverse(root);
        sort(ans.begin(),ans.end());
        return ans[k-1];
    }
    void traverse(TreeNode* r){
        if(r==NULL)
            return;
         ans.push_back(r->val);
         traverse(r->left);
        traverse(r->right);
    }
};

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        
        self.li=[]
        def search(root):
            if not root:
                return 
            self.li.append(root.val)
            search(root.right)
            search(root.left)
        search(root)
        self.li.sort()
        return self.li[k-1] 